it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/errorControllers/CannotConfirmBusinessControllerISpec.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers.trustControllers.errorControllers
+
+import play.api.libs.ws.WSResponse
+import play.api.test.Helpers.{NO_CONTENT, OK, SEE_OTHER, await, defaultAwaitTimeout}
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants.{testContinueUrl, testInternalId, testJourneyId, testTrustsJourneyConfig}
+import uk.gov.hmrc.minorentityidentificationfrontend.controllers.trustControllers.{routes => trustRoutes}
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, StorageStub}
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ComponentSpecHelper
+import uk.gov.hmrc.minorentityidentificationfrontend.views.trustViews.CannotConfirmBusinessViewTests
+
+class CannotConfirmBusinessControllerISpec extends ComponentSpecHelper
+  with AuthStub
+  with StorageStub
+  with CannotConfirmBusinessViewTests {
+
+  "GET /cannot-confirm-business" should {
+    lazy val result = {
+      await(journeyConfigRepository.insertJourneyConfig(
+        journeyId = testJourneyId,
+        authInternalId = testInternalId,
+        journeyConfig = testTrustsJourneyConfig(true)
+      ))
+      stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+      get(s"/identify-your-trust/$testJourneyId/cannot-confirm-business")
+    }
+
+    "return OK" in {
+      result.status mustBe OK
+    }
+
+    "return a view which" should {
+      testCannotConfirmBusinessView(result)
+    }
+
+    "redirect to sign in page" when {
+      "the user is UNAUTHORISED" in {
+        stubAuthFailure()
+        lazy val result: WSResponse = get(s"/identify-your-trust/$testJourneyId/cannot-confirm-business")
+        result must have(
+          httpStatus(SEE_OTHER),
+          redirectUri("/bas-gateway/sign-in" +
+            s"?continue_url=%2Fidentify-your-trust%2F$testJourneyId%2Fcannot-confirm-business" +
+            "&origin=minor-entity-identification-frontend"
+          )
+        )
+      }
+    }
+  }
+
+  "POST /cannot-confirm-business" when {
```

Comment:
I would add a test for unauthorized users.

Reply:
Done